File: 1706-min-cost-to-connect-all-points/min-cost-to-connect-all-points.py
class Solution:
    def minCostConnectPoints(self, points: List[List[int]]) -> int:
        visited = set()
        min_heap = [(0, points[0][0], points[0][1])]
        total_cost = 0
        while min_heap:
            cost, i, j = heapq.heappop(min_heap)
            if (i, j) in visited:
                continue
            if len(visited) == len(points):
                break
            total_cost += cost
            visited.add((i, j))
            for next_i, next_j in points:
                if (next_i, next_j) in visited:
                    continue
                cost = abs(next_i - i) + abs(next_j - j)
                heapq.heappush(min_heap, (cost, next_i, next_j))
        return total_cost

File: 1706-min-cost-to-connect-all-points/min-cost-to-connect-all-points.java
class Solution {
    public int minCostConnectPoints(int[][] points) {
      int mstCost = 0;
      PriorityQueue<Integer[]> minHeap = new PriorityQueue<>((a, b) -> a[0] - b[0]);
      Set<Integer> visited = new HashSet<>();
      Integer[] initPoint = {0, 0};
      minHeap.add(initPoint);
      while (!minHeap.isEmpty()) {
        Integer[] data = minHeap.poll();
        int distance = data[0];
        int i = data[1];
        int[] currPoint = points[i];
        if (visited.contains(i)) {
            continue;
        } 
        visited.add(i);
        mstCost += distance;
        for (int j = 0; j < points.length; j++) {
            if (visited.contains(j)) {
                continue;
            }
            int[] nextPoint = points[j];
            int nextDistance = Math.abs(currPoint[0] - nextPoint[0]) + Math.abs(currPoint[1] - nextPoint[1]);
            Integer[] nextData = {nextDistance, j};
            minHeap.add(nextData);
        }
        
      }
      return mstCost;
    }
}